### ISSUE TITLE
# Database Schema:

### ISSUE THREAD

**Issue by TahjDev** [avakil3/Stopify#2]
- [x] Contains correct datatypes
- [x] Contains appropriate constraints/details
    - [x] primary key
    - [x] not null
    - [x] unique
    - [x] indexed
    - [x] foreign key
- [ ] Contains bullet points after the table that state which foreign keys will reference to which table, or references to the associations which will be made
- [x] foreign key and table name are lowercased, snake_cased and back_ticked
- [x] Correctly formatted
    - [x] schema is written in a table format
    - [x] the table's name are lowercased, snake_cased and back_ticked
    - [x] the table header column names are bolded
    - [x] columns names are lowercased and snaked_cased and back_ticked

**Comment by TahjDev** [comment]
- Inside of your schema in your details, I see you have present under details what is that referencing? I think the not null may be taking care of that 
- Under all the tables can you list what the foreign_keys are referencing your polymorphic table as well
- Is there a difference between liked items and a playlist, if so then I believe you would need a playlist table to handle that relationship between playlists and songs

**Comment by avakil3** [comment]
Response to Bullet #1: The 'present' tag under details is referring to the model level validation of "presence: true" as compared to the database level constraint of "null:false". If you think that is duplicative, I can remove the tag "present".

Response to Bullet #2: I added bullet points under each table referencing the foreign keys and added foreign keys for the polymorphic associations

Response to Bullet #3: There is a difference between liked items and playlists. Liked items will appear under a "library' page as users can like songs, albums, or artists. I added an additional "Playlist_Songs" table to address the relationship between playlists and their songs.
